### PR TITLE
Improve TensorFlow.js performance on mobile

### PR DESCRIPTION
**Pull request by JakeTurner616**
### 1. **Performance Optimizations**

* Introduced a Web Worker-based inference system (`frameWorker.ts`) to run TensorFlow\.js off the main thread.
* Replaced direct tensor inference with `createImageBitmap` + async message passing.
* Added `MIN_FRAME_INTERVAL_MS` to cap frame processing rate and avoid UI thread blocking.
* Removed `opencv.worker.ts` — has not been used for a long time.

---

### 2. **TensorFlow\.js Backend Integration**

* Added and configured `@tensorflow/tfjs-backend-wasm` for WebAssembly inference.
* Bundled `.wasm` binaries in `public/` for TFJS runtime compatibility.

---

### 3. **Model Shrinkage**

* Swapped out the 13-shard model for a smaller 6-shard export.
* Dramatically improves load time and memory usage in mobile browsers.

---

### 4. **UI + Rendering Updates**

* Refactored `CameraStream` and `CameraPanel` to show the `<video>` element alongside canvas overlay.
* Quad rendering now respects confidence thresholds to suppress noisy predictions.

---

### 5. **React Hook Refactor**

* `useFrameProcessor` no longer imports or runs TFJS directly.
* All inference is now performed inside the worker, managed via `frameWorkerSingleton.ts`.

---

#### **Model Architecture**

* **Before**: ResNet50 + skip connections (heavy, high-accuracy)
* **Now**: MobileNetV2 (lightweight, optimized for mobile browser w/ webgl)

#### **Output Resolution**

* Heatmaps: **112×112 → 56×56**
* Sigma: **2.0 → 1.5** (tighter keypoint peaks)

#### **Training Simplification**

* Removed heavy callbacks (e.g. TQDM, ReduceLROnPlateau)
* Export now uses `model.export()` for WASM-safe SavedModel

#### **Web Deployment**

* Old: `.keras → SavedModel → TFJS` (manual)
* New: direct export to `tfjs_graph_model` (WASM-compatible although webgl is mainly used)
* Model shards reduced: **13 → 6**

#### **Runtime Impact**

* Inference: **\~8–10 FPS** on mobile (tested on an iphone 7 up from \~1–2)
* Smaller model, faster load, similar prediction accuracy

As always the [training workflow notebook](https://www.kaggle.com/code/jaketurner616/mtg-keypoint-regression-heatmap) and [keypoint regression repo](https://github.com/JakeTurner616/simple-mtg-keypoint-regression) reflect these changes

### CODE REVIEW

File: vite-frontend/src/hooks/useFrameProcessor.ts [hunk 1]
```diff
@@ -110,229 +114,245 @@ const useFrameProcessor = ({
...
-      !videoRef.current ||
-      !canvasRef.current ||
-      !modelLoaded.current ||
-      !modelRef.current
-    ) {
-      raf.current = requestAnimationFrame(process);
-      return;
-    }
-
+  const interval = setInterval(() => {
```
[the remark this answers is not in the document]
Comment (JakeTurner616) [reply vite-frontend/src/hooks/useFrameProcessor.ts:123]: fixed in [a4756](https://github.com/mtgscan-cards/monolithic/pull/6/commits/a475692c0a96c6ff56a83324b79a261a94d8b536)
